packages/react/src/components/ErrorBoundary/ErrorBoundary-story.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState } from 'react';
+import { storiesOf } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+import { ErrorBoundary, ErrorBoundaryContext } from './';
+import Button from '../Button';
+
+storiesOf('ErrorBoundary', module)
+  .add('default', () => {
+    function DemoComponent() {
+      const [shouldThrowError, setShouldThrowError] = useState(false);
+
+      function onClick() {
+        setShouldThrowError(!shouldThrowError);
+      }
+
+      return (
+        <>
+          <Button onClick={onClick}>Toggle throwing error</Button>
+          <div>
+            <ErrorBoundary fallback={<Fallback />}>
+              <ThrowError shouldThrowError={shouldThrowError} />
+            </ErrorBoundary>
+          </div>
+        </>
+      );
+    }
+
+    function Fallback() {
+      return 'Whoops';
+    }
+
+    function ThrowError({ shouldThrowError }) {
+      if (shouldThrowError) {
+        throw new Error('Component threw error');
+      }
+
+      return 'Successfully rendered';
+    }
+
```

Comment:
@joshblack You could add this to give `DemoComponent` the same info as `ErrorBoundary`, so it will show the right component name in the story source + also show the props in the prop table.

(Though unfortunately all props will be `node` type -- I'm not sure how to fix that yet)

```suggestion

    DemoComponent.displayName = 'ErrorBoundary';
    DemoComponent.__docgenInfo = {
      ...ErrorBoundary.__docgenInfo,
      props: {
        ...ErrorBoundary.__docgenInfo.props,
      },
    };
```

Reply:
This might be a dumb question, but what's the difference between this proposal and:

```js
DemoComponent.__docgenInfo = ErrorBoundary.__docgenInfo;
```

Just noticed the double spread and was curious.

Reply:
Oh there's no difference! I think I've personally gotten into the habit of doing this as a way to very explicitly show "I'm need the props here" etc 😆 I'll update my suggestions to be more succinct 👍